code/datums/weather/weather.dm

```diff
@@ -105,6 +105,10 @@
 	update_areas()
 
 /datum/weather/proc/can_weather_act(mob/living/L) //Can this weather impact a mob?
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		if(H.get_thermal_protection() >= FIRE_IMMUNITY_SUIT_MAX_TEMP_PROTECT)
```

Comment:
this belongs in ash storm weather, not `/datum/weather`

Reply:
`can_weather_act` existed in weather.dm before I made any changes. If I make a separate `can_weather_act` in ash_storm.dm, is it going to cause issues?

Reply:
no because (using the code i wrote earlier) it calls inher.

Reply:
If I move the whole thing into Ash_storm, it's going to break other storms, right? So what should I do?

Reply:
So I reverted the weather.dm file and added `can_weather_act` in ash_storm.dm as well, and now the storms damage nothing whatsoever. weather.dm had a `can_weather_act` section too, so I'm guessing it's a conflict?

Reply:
you do not touch this proc and copy what I put in 

```DM
/datum/weather/ash_storm/can_weather_act(mob/living/L)
    if(ishuman(L))
        var/mob/living/carbon/human/H = L
        if(H.get_thermal_protection() >= FIRE_IMMUNITY_SUIT_MAX_TEMP_PROTECT)
            return FALSE
    ..() //weather_immunities etc etc
```